command/v6/add_network_policy_command_test.go

```diff
@@ -150,11 +160,99 @@ var _ = Describe("add-network-policy Command", func() {
 				Expect(executeErr).ToNot(HaveOccurred())
 
 				Expect(fakeActor.AddNetworkPolicyCallCount()).To(Equal(1))
-				_, _, _, passedProtocol, passedStartPort, passedEndPort := fakeActor.AddNetworkPolicyArgsForCall(0)
+				_, _, _, _, passedProtocol, passedStartPort, passedEndPort := fakeActor.AddNetworkPolicyArgsForCall(0)
 				Expect(passedProtocol).To(Equal("tcp"))
 				Expect(passedStartPort).To(Equal(8080))
 				Expect(passedEndPort).To(Equal(8080))
 			})
 		})
+
+		When("org is specified and space is not", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = ""
+			})
+
+			It("returns an error", func() {
+				Expect(executeErr).To(MatchError(translatableerror.NetworkPolicyDestinationOrgWithoutSpaceError{}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
+			})
+		})
+
+		When("invalid org is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{}, v3action.Warnings{}, actionerror.OrganizationNotFoundError{Name: "bananarama"})
+			})
+
+			It("returns an error", func() {
+				Expect(executeErr).To(MatchError(actionerror.OrganizationNotFoundError{Name: "bananarama"}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
+			})
+		})
+
+		When("a valid org but invalid space is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationOrg = "bananarama"
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetOrganizationByNameReturns(v3action.Organization{GUID: "some-org-guid"}, v3action.Warnings{}, nil)
+				fakeActor.GetSpaceByNameAndOrganizationReturns(v3action.Space{}, v3action.Warnings{}, actionerror.SpaceNotFoundError{Name: "bananarama"})
+			})
+
+			It("returns an error", func() {
+				passedSpaceName, passedOrgGuid := fakeActor.GetSpaceByNameAndOrganizationArgsForCall(0)
+				Expect(passedSpaceName).To(Equal("hamdinger"))
+				Expect(passedOrgGuid).To(Equal("some-org-guid"))
+				Expect(executeErr).To(MatchError(actionerror.SpaceNotFoundError{Name: "bananarama"}))
+				Expect(testUI.Out).NotTo(Say(`Adding network policy`))
+			})
+		})
+
+		When("a destination space but no destination org is specified", func() {
+			BeforeEach(func() {
+				cmd.DestinationSpace = "hamdinger"
+				fakeActor.GetSpaceByNameAndOrganizationReturns(v3action.Space{GUID: "some-other-space-guid"}, v3action.Warnings{}, nil)
```

Comment:
Please add warnings here and corresponding assertions testing that those warnings are printed.

Reply:
😸